src/server/scripts/EasternKingdoms/Stratholme/boss_maleki_the_pallid.cpp

```diff
@@ -55,15 +63,12 @@ class boss_maleki_the_pallid : public CreatureScript
 
         InstanceScript* instance;
 
-        uint32 Frostbolt_Timer;
-        uint32 IceTomb_Timer;
-        uint32 DrainLife_Timer;
-
         void Reset() override
         {
-            Frostbolt_Timer = 1000;
-            IceTomb_Timer = 16000;
-            DrainLife_Timer = 31000;
+            _events.Reset();
+            _events.ScheduleEvent(EVENT_FROSTBOLT,  1 * IN_MILLISECONDS);
+            _events.ScheduleEvent(EVENT_ICETOMB,   16 * IN_MILLISECONDS);
+            _events.ScheduleEvent(EVENT_DRAINLIFE, 31 * IN_MILLISECONDS);
```

Comment:
Events are scheduled on EnterCombat() and reset on Reset(). Did you test this in-game and did it work properly?